File: pat1101.py
"""
    attention: test case 1 Time Limit Exceeded
"""

n = int(input())
nums = [int(x) for x in input().split()]
nums_sorted = sorted(nums)
pivot_candidates = []
max_num = 0
for i in range(n):
    if nums[i] == nums_sorted[i] and nums[i] > max_num:
        pivot_candidates.append(nums[i])
    if nums[i] > max_num:
        max_num = nums[i]

print(len(pivot_candidates))
if pivot_candidates:
    print(pivot_candidates[0], end='')
    for v in pivot_candidates[1:]:
        print(" " + str(v), end='')
else:
    print()

File: pat1101.java
// attention: test case 1,2,3,4,5 Time Limit Exceeded

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine());
        String[] line = reader.readLine().split(" ");
        int[] nums = new int[n];
        int[] numsSorted = new int[n];
        for (int i = 0; i < n; i++) {
            int temp = Integer.parseInt(line[i]);
            nums[i] = temp;
            numsSorted[i] = temp;
        }

        Arrays.sort(numsSorted);
        List<Integer> results = new ArrayList<>();

        int maxNum = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == numsSorted[i] && nums[i] > maxNum) {
                results.add(nums[i]);
            }
            if (nums[i] > maxNum) {
                maxNum = nums[i];
            }
        }

        System.out.println(results.size());
        if (results.size() != 0) {
            System.out.print(results.get(0));
            for (int i = 1; i < results.size(); i++) {
                System.out.print(" " + results.get(i));
            }
        }else {
            System.out.println();
        }

    }
}
